msgraphcore/graph_session.py

```diff
@@ -27,9 +29,11 @@ def __init__(
         self._base_url = BASE_URL + '/' + api_version
 
         auth_handler = AuthorizationHandler(credential, scopes)
+        retry_handler = RetryMiddleware(retry_config)
 
         # The authorization handler should be the first middleware in the pipeline.
         middleware.insert(0, auth_handler)
+        middleware.insert(1, retry_handler)
```

Comment:
I think we should consider adding a final HTTPAdapter to the pipeline after the GraphClientFactory creates the default pipeline or after a customer provides their own pipeline. This final handler should optionally set a proxy address if provided when initializing `GraphSession`.

Reply:
An issue has been created for that https://github.com/microsoftgraph/msgraph-sdk-python-core/issues/80